### DISCUSSION TITLE
support for Mistral ?

### DISCUSSION THREAD

**Discussion by HoomerPH** [cline/cline#685] [Feature Requests]
i wish we have support on Mistral AI and Codestral, using openrouter is such a hassle, why not pasting the api key on the go right ?

**Comment by RaySinner** [comment]
Mistral has released an update to their **free** Pixtral and Codestral models. I would like to see this provider in the list of available ones.
![image](https://github.com/user-attachments/assets/ca974e49-72de-4607-9991-059fc31c26c4)
![image](https://github.com/user-attachments/assets/5d030329-2313-4d17-8690-61368fd87b34)
Maybe I don’t understand something, but I have a free plan and I calmly use Pixtral in “Continue” and 1B free tokens per month for a model at the 128k & level score of ChatGPT-4o seems like just a birthday present. The only problem is the limit of 1 request per second, which needs to be taken in your future connector @saoudrizwan

**Comment by jhmonroe** [comment]
Came here to ask same question — would love to test out CODESTRAL (Mistral's LLM specifically for coding) within Cline.

Mistral currently allows free API access for now, including to the Codestral API specifically and its own endpoints:
completion: https://codestral.mistral.ai/v1/fim/completions
chat: https://codestral.mistral.ai/v1/chat/completions

https://mistral.ai/news/codestral/
More info on setup: https://www.datacamp.com/tutorial/codestral-api-tutorial

**Comment by uschuck** [comment]
the same here,  we would love to see mistral in the API Providers .

**Comment by zlocate** [comment]
@HoomerPH you may temporarily use the LiteLLM proxy to start an OpenAI-compatible proxy API and use it with Cline.

---

### Steps:

1. Go to Cline settings and select the **OpenAI-compatible API** as shown in the screenshot below:

![image](https://github.com/user-attachments/assets/1e8f0033-f1ed-4210-9872-ec8403ce0d23)


2. **Install LiteLLM**:  
```bash
pip install litellm litellm[proxy]
```

Export the API key and start the LiteLLM server:

```bash
export MISTRAL_API_KEY=your_api_key
litellm --model mistral/mistral-large-latest
```
Replace `mistral/mistral-large-latest` with the name of the required Codestral model.

**Comment by dcbartlett** [comment]
Mistral Codestral has been supported for a while now.  Closing this as completed.
![image](https://github.com/user-attachments/assets/65b5ce0c-801a-4ec9-8404-3c8d633f653b)
